src/Uno.UI/Media/PathMarkupParser.cs

```diff
@@ -41,426 +39,574 @@ namespace Uno.Media
-                                ReadSeparator(reader);
-                                SweepDirection sweepDirection = ReadBool(reader) ? SweepDirection.Clockwise : SweepDirection.Counterclockwise;
-                                ReadSeparator(reader);
-                                point = ReadPoint(reader, point, relative);
-
-                                _context.ArcTo(point, size, rotationAngle, isLargeArc, sweepDirection, true, false);
-                                previousControlPoint = null;
-                                break;
-                            }
-
-                        case Command.Close:
-                            _context.SetClosedState(true);
-                            openFigure = false;
-                            previousControlPoint = null;
-                            break;
-
-                        default:
-                            throw new NotSupportedException("Unsupported command");
-                    }
-                }
-
-                if (openFigure)
-                {
-                    _context.SetClosedState(false);
-                }
-            }
-        }
-
-        private Point MirrorControlPoint(Point controlPoint, Point center)
-        {
-            Point dir = (controlPoint - center);
-            return center - dir;
-        }
-
-        private static bool ReadCommand(
-            StringReader reader,
-            ref Command command,
-            ref bool relative)
-        {
-			// https://www.w3.org/TR/SVG/paths.html#PathDataMovetoCommands
-			// If a moveto is followed by multiple pairs of coordinates, the subsequent pairs are treated as implicit lineto commands
-			if (command == Command.Move)
+		private static readonly Dictionary<char, Command> s_commands =
+		   new Dictionary<char, Command>
+			   {
+					{ 'F', Command.FillRule },
+					{ 'M', Command.Move },
+					{ 'L', Command.Line },
+					{ 'H', Command.HorizontalLine },
+					{ 'V', Command.VerticalLine },
+					{ 'Q', Command.QuadraticBezierCurve },
+					{ 'T', Command.SmoothQuadraticBezierCurve },
+					{ 'C', Command.CubicBezierCurve },
+					{ 'S', Command.SmoothCubicBezierCurve },
+					{ 'A', Command.Arc },
+					{ 'Z', Command.Close },
+			   };
+
+		// Uno specific: Use StreamGeomoetryContext instead of IGeometryContext.
+		private StreamGeometryContext _geometryContext;
+		private Point _currentPoint;
+		private Point? _beginFigurePoint;
+		private Point? _previousControlPoint;
+		private bool _isOpen;
+		private bool _isDisposed;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="PathMarkupParser"/> class.
+		/// </summary>
+		/// <param name="geometryContext">The geometry context.</param>
+		/// <exception cref="ArgumentNullException">geometryContext</exception>
+		public PathMarkupParser(StreamGeometryContext geometryContext) // Uno specific: Use StreamGeomoetryContext instead of IGeometryContext.
```

Comment:
CI is likely to fail due to parameter name of public API. @jeromelaban Is this "very very" small breaking change (likely to affect near 0 customers) okay?, or should I just diverge from Avalonia source here and keep the original parameter name?

Reply:
No breaking changes are better, but I see that you've changed it.